lib/src/subtree.c

```diff
@@ -773,6 +774,10 @@ Subtree ts_subtree_edit(Subtree self, const TSInputEdit *input_edit, SubtreePool
       ) && (
         !invalidate_first_row ||
         child_left.extent.row > entry.tree->ptr->padding.extent.row
+      ) && (
+      	!ts_subtree_depends_on_column(*child) ||
+        !column_shifted ||
+        child_left.extent.row > edit.old_end.extent.row
```

Comment:
This new condition makes good sense. Did you check if you can actually *remove* the previous condition? I would think that you'd only need to check `depends_on_column` in one spot. Do any tests fail if you remove the second part of this 3-part `&&` expression?

Reply:
> Did you check if you can actually _remove_ the previous condition?

Yea I did, and I can't: it would make the Haskell-like indentation test fail -- I originally thought that it's the test that were wrong.

> I would think that you'd only need to check `depends_on_column` in one spot.

That would be the case if trees that depend-on-column make their supertree depend-on-column too. But that's not the case because currently the condition for setting the flag is:
```c
// subtree.c line 394
    if (
      self.ptr->size.extent.row == 0 &&
      ts_subtree_depends_on_column(child)
    ) {
      self.ptr->depends_on_column = true;
    }
```
So it's possible for a tree to NOT depend on column, but it contains subtree that DOES. (I find that counterintuitive, but maybe there's a reason for that?).
That's why we need to check `depends_on_column` on two spots: one when iterating the bisecting-stack, and another when delving into the children -- the condition needs to be re-checked for each subtree.

Now, I did try changing the flag-setting code to:
```c
// subtree.c line 394
    if (
      ts_subtree_depends_on_column(child)
    ) {
      self.ptr->depends_on_column = true;
    }
```
- All the tests still pass after this change.
- It's more intuitive: if a tree depends-on-column, then its supertree depends-on-column too.
- On subtree-edit, we just need to check the flag at one spot only.

But is it OK? If so, I'll make the change.

> Do any tests fail if you remove the second part of this 3-part `&&` expression?

No, all tests still pass without the second condition -- it's an optimization for when the column positions does not change e.g. if the length of `inserted_text` == `deleted_length`. Do you think it's incorrect/premature? Or is it ok but I should add a test for that (in the form of a recorder reading fewer bytes)?

Reply:
No, then almost the entire tree would depend on its column.

A tree only depends on its start column of any of its children **that start on its first line** depend on their column. For all subsequent children, the tree's start column does not change their column.